pkg/kv/kvserver/replica_init.go

```diff
@@ -232,11 +233,13 @@ func newUninitializedReplicaWithoutRaftGroup(
 	r.raftMu.msgAppScratchForFlowControl = map[roachpb.ReplicaID][]raftpb.Message{}
 	r.raftMu.replicaStateScratchForFlowControl = map[roachpb.ReplicaID]rac2.ReplicaStateInfo{}
 	r.flowControlV2 = replica_rac2.NewProcessor(replica_rac2.ProcessorOptions{
-		NodeID:                 store.NodeID(),
-		StoreID:                r.StoreID(),
-		RangeID:                r.RangeID,
-		ReplicaID:              r.replicaID,
-		Replica:                (*replicaForRACv2)(r),
+		NodeID:            store.NodeID(),
+		StoreID:           r.StoreID(),
+		RangeID:           r.RangeID,
+		ReplicaID:         r.replicaID,
+		ReplicaForTesting: (*replicaForRACv2)(r),
```

Comment:
Does this need to be non-nil? It seems to be used only in tests (and right now the only method call is hidden behind a knob check which will never let this call to happen in prod code).

Leaving it non-nil seems safer though, in case the method is called by mistake.